### PR TITLE
Add support for WebP

### PR DESCRIPTION
**Pull request by levibuzolic**
https://developers.google.com/speed/webp/docs/riff_container

There are 3 WebP formats, each with their own method of encoding width and height.

 * Lossy: `VP8`
 * Lossless: `VP8L`
 * Extended: `VP8X`

### CODE REVIEW

File: types/webp.js [hunk 1]
```diff
@@ -0,0 +1,62 @@
+const isWebp = bytes =>
+	bytes[0] === 0x52
+	&& bytes[1] === 0x49
+	&& bytes[2] === 0x46
+	&& bytes[3] === 0x46
+	&& bytes[8] === 0x57
+	&& bytes[9] === 0x45
+	&& bytes[10] === 0x42
+	&& bytes[11] === 0x50;
```
Comment (levibuzolic) [inline types/webp.js:9]:
```
 0                   1                   2                   3
 0 1 2 3 4 5 6 7 8 9 0 1 2 3 4 5 6 7 8 9 0 1 2 3 4 5 6 7 8 9 0 1
+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
|      'R'      |      'I'      |      'F'      |      'F'      |
+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
|                           File Size                           |
+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
|      'W'      |      'E'      |      'B'      |      'P'      |
+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
```

File: types/webp.js [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+	if (!isWebp(bytes)) {
+		return;
+	}
+
+	const maxSize = 0x3F_FF;
+
+	if (isVP8Lossy(bytes)) {
+		return {
+			/* eslint-disable no-bitwise */
+			width: readUInt16LE(bytes, 26) & maxSize,
```
Comment (sindresorhus) [inline types/webp.js:50]: Can you use `DataView` instead?

File: types/webp.js [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+			height: readUInt16LE(bytes, 28) & maxSize,
```
Comment (sindresorhus) [inline types/webp.js:51]: You are sure the size is at a fixed offset always? Just making sure.

**Comment by sindresorhus** [comment]
Thanks for working on this 👍

Can you update the readme too?

**Review by sindresorhus — CHANGES_REQUESTED** [review]

File: types/webp.js (hunk 3 above)
Comment (levibuzolic) [reply types/webp.js]: Yes I beleive so, `VP8` and `VP8L` both have a firxt 12 byte file header. The frame data then has a fixed header layout with a slightly different format between the lossy and lossless encoding: https://datatracker.ietf.org/doc/html/rfc6386#autoid-15

**Comment by levibuzolic** [comment]
Thanks for the feedback @sindresorhus I think this should be good to go now.

**Comment by sindresorhus** [comment]
Nice work! Thanks for contributing this 🙏

**Pull request merged**